src/login/logind.c

```diff
@@ -328,11 +328,18 @@ static int manager_enumerate_users(Manager *m) {
 
         FOREACH_DIRENT(de, d, return -errno) {
                 User *u;
+                uid_t uid;
 
                 if (!dirent_is_file(de))
                         continue;
 
-                k = manager_add_user_by_name(m, de->d_name, &u);
+                k = safe_atou(de->d_name, &uid);
```

Comment:
oh, yikes! what an awful bug!

but please use parse_uid() here!

Reply:
Ah sorry wanted to use parse_uid() but for some reason I expected to find it in parse-util.h. Since it was not there I came up with my own. Updated.